package/MDAnalysis/core/topologyattrs.py

```diff
@@ -1243,38 +1340,43 @@ class SegmentAttr(TopologyAttr):
     """
     attrname = 'segmentattrs'
     singular = 'segmentattr'
-    target_classes = [Segment]
+    target_classes = [AtomGroup, ResidueGroup, SegmentGroup, Segment]
     per_object = 'segment'
 
     def get_atoms(self, ag):
-        six = self.top.tt.atoms2segments(ag._ix)
+        six = self.top.tt.atoms2segments(ag.ix)
```

Comment:
why did you make this variable public?

Reply:
it's not a new attribute, it's just we shouldn't use the "private" version of our own attributes